### PR TITLE
|CM_APP_57|Sending Notification for recover CM id

### PR DESCRIPTION
**Pull request by RajeswariTW**
Please review

### CODE REVIEW

File: src/In.ProjectEKA.OtpService/Notification/NotificationService.cs [hunk 1]
```diff
@@ -15,14 +15,23 @@ public NotificationService(ISmsClient smsClient)
...
-            };
+                case Action.ConsentRequestCreated:
+                    await smsClient.Send(
+                        notification.Communication.Value,
+                        GenerateConsentRequestMessage(notification.Content));
+                    return new Response(ResponseType.InternalServerError, "");
+
+                case Action.CmidRecovered:
+                    await smsClient.Send(
+                        notification.Communication.Value,
```
Comment (ganesanarun) [inline src/In.ProjectEKA.OtpService/Notification/NotificationService.cs]: You are supposed to call another method.

File: src/In.ProjectEKA.OtpService/Notification/Action.cs [hunk 2]
```diff
@@ -2,6 +2,6 @@ namespace In.ProjectEKA.OtpService.Notification
 {
     public enum Action
     {
-        ConsentRequestCreated
+        ConsentRequestCreated, CmidRecovered
```
Comment (ganesanarun) [inline src/In.ProjectEKA.OtpService/Notification/Action.cs]: ConsentManagerAccountRecovered in the next line

File: src/In.ProjectEKA.OtpService/Notification/NotificationService.cs [hunk 3]
```diff
@@ -34,5 +43,14 @@ private static string GenerateConsentRequestMessage(JObject notificationContent)
                 $"To view request, please tap on the link: {content.DeepLinkUrl}";
             return message;
         }
+
+        private static string GenerateCmidRecoveryMessage(JObject notificationContent)
+        {
+            var cMidContent = notificationContent.ToObject<CMidContent>();
```
Comment (ganesanarun) [inline src/In.ProjectEKA.OtpService/Notification/NotificationService.cs]: Why M caps here? Then should not your CmidRecovered =› be CMidRecovered, why id is not treated as disjointed to the word CM

File: src/In.ProjectEKA.OtpService/Notification/NotificationService.cs [hunk 4]
```diff
@@ -15,14 +15,23 @@ public NotificationService(ISmsClient smsClient)
         }
 
         public async Task<Response> SendNotification(Notification notification)
-        {
-            return notification.Action switch
+        { 
+            switch(notification.Action)
```
Comment (ganesanarun) [inline src/In.ProjectEKA.OtpService/Notification/NotificationService.cs]: What was wrong with pattern matching?

File: src/In.ProjectEKA.OtpService/Notification/Action.cs [hunk 5]
```diff
@@ -2,6 +2,6 @@ namespace In.ProjectEKA.OtpService.Notification
 {
     public enum Action
     {
```
Comment (ganesanarun) [inline src/In.ProjectEKA.OtpService/Notification/Action.cs:4]: Don't you feel like we should have test for this?

Comment (RajeswariTW) [reply src/In.ProjectEKA.OtpService/Notification/Action.cs:4]: I am not sure... to the NotificationService.SendNotification?

File: src/In.ProjectEKA.OtpService/Notification/NotificationService.cs (hunk 1 above)
Comment (RajeswariTW) [reply src/In.ProjectEKA.OtpService/Notification/NotificationService.cs]: Corrected

File: src/In.ProjectEKA.OtpService/Notification/Action.cs (hunk 2 above)
Comment (RajeswariTW) [reply src/In.ProjectEKA.OtpService/Notification/Action.cs]: Refactor the code where ever is required

File: src/In.ProjectEKA.OtpService/Notification/NotificationService.cs (hunk 3 above)
Comment (RajeswariTW) [reply src/In.ProjectEKA.OtpService/Notification/NotificationService.cs]: Refactor the code where ever is required

File: src/In.ProjectEKA.OtpService/Notification/NotificationService.cs (hunk 4 above)
Comment (RajeswariTW) [reply src/In.ProjectEKA.OtpService/Notification/NotificationService.cs]: corrected

File: src/In.ProjectEKA.OtpService/Notification/Action.cs (hunk 5 above)
Comment (RajeswariTW) [reply src/In.ProjectEKA.OtpService/Notification/Action.cs:4]: Added tests

**Pull request merged**